tests/flight/sync.rs

```diff
@@ -609,3 +609,70 @@ async fn test_sync_custom_store(
 
     Ok(())
 }
+
+#[tokio::test]
+async fn test_sync_iceberg_custom_store(
```

Comment:
Seems like `test_sync_custom_store` and this test share the setup/results, maybe we can just parametrize the former to do exercise this routine for both formats (we can ditch the `local` parametrization in it since it won't work for Iceberg)?